src/hotspot/share/runtime/nonJavaThread.cpp

```diff
@@ -270,8 +270,8 @@ void WatcherThread::run() {
           os::die();
         }
 
-        // Wait a second, then recheck for timeout.
-        os::naked_short_sleep(999);
```

Comment:
Harmless change but I don't see why we need sub-second resolution when the ErrorLogTimeout is in seconds. ??

Reply:
This matters because it also applies to each step timeout too. If we check only once per second, we overshoot each timeout by up to one second. This overshooting happens for every step timeout. If we are in a situation like here where we ignore the global step timeout, and we keep running into deadlocks e.g. in malloc, we will encounter a lot of step timeouts; these would add up.

Reply:
Note that this matters even more in the context of https://github.com/openjdk/jdk/pull/11017, since that will increase the granularity for each test, potentially exposing us to a lot more individual timeouts.

Reply:
I thought there was a limit of three step timeouts before we gave up? And the step timeout is a quarter of the global so still many seconds.